src/c/c_dataset.cpp

```diff
@@ -30,163 +30,196 @@
-                        void** data,
-                        size_t** dims,
-                        size_t* n_dims,
-                        CTensorType* type,
-                        CMemoryLayout mem_layout)
+                        void* *data,
+                        size_t* *dims,
+                        size_t *n_dims,
+                        CTensorType *type,
+                        const CMemoryLayout mem_layout)
 {
-  /* Get a tensor of a specified type from the database.
-  This function may allocate new memory for the tensor.
-  This memory will be deleted when the user deletes the
-  DataSet object.
-  */
-  DataSet* d = (DataSet*)dataset;
-  std::string name_str = std::string(name, name_length);
-
-  TensorType t_type;
+  // Sanity check parameters
+  if (NULL == dataset || NULL == name || NULL == data ||
+      NULL == dims || NULL == n_dims || NULL == type)
+    return;
+
+  DataSet *d = (DataSet *)dataset;
+  std::string name_str(name, name_length);
+
+  TensorType t_type = TensorType::undefined;
   d->get_tensor(name_str, *data, *dims, *n_dims,
                 t_type, convert_layout(mem_layout));
   *type = convert_tensor_type(t_type);
-  return;
 }
 
+// Copy the tensor data buffer into the provided memory space (data).
 extern "C"
-void unpack_dataset_tensor(void* dataset,
-                           const char* name,
+void unpack_dataset_tensor(void *dataset,
+                           const char *name,
                            const size_t name_length,
-                           void* data,
-                           const size_t* dims,
+                           void *data,
+                           const size_t *dims,
                            const size_t n_dims,
                            const CTensorType type,
                            const CMemoryLayout mem_layout)
 {
-  /* This function will take the tensor data buffer and copy
-  it into the provided memory space (data).
-  */
-  DataSet* d = (DataSet*)dataset;
-  std::string name_str = std::string(name, name_length);
+  // Sanity check parameters
+  if (NULL == dataset || NULL == name || NULL == data || NULL == dims)
+    return;
+
+  DataSet *d = reinterpret_cast<DataSet *>(dataset);
+  std::string name_str(name, name_length);
 
   std::vector<size_t> dims_vec;
-  for(size_t i=0; i<n_dims; i++)
-    dims_vec.push_back(dims[i]);
+  dims_vec.assign(dims, dims + n_dims);
 
   d->unpack_tensor(name_str, data, dims_vec,
                    convert_tensor_type(type),
                    convert_layout(mem_layout));
-
-  return;
 }
 
+// Get a meta data field. This method may allocate
+// memory that is cleared when the user deletes the
+// DataSet object
 extern "C"
-void* get_meta_scalars(void* dataset,
-                      const char* name,
-                      const size_t name_length,
-                      size_t* length,
-                      CMetaDataType* type)
+void *get_meta_scalars(void *dataset,
+		       const char *name,
```

Comment:
Maybe fix alignment so that all types are in vertical alignment.

Reply:
definitely. Not sure what happened here, probably a tab/space issue